app/src/main/java/net/gsantner/markor/ui/SearchOrCustomTextDialogCreator.java

```diff
@@ -442,6 +446,27 @@ public static void showPriorityDialog(Activity activity, char selectedPriority,
         SearchOrCustomTextDialog.showMultiChoiceDialogWithSearchFilterUI(activity, dopt);
     }
 
+    public static void showSetPasswordDialog(final Activity activity) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            final AppSettings as = new AppSettings(activity.getApplicationContext());
+            final SearchOrCustomTextDialog.DialogOptions dopt = new SearchOrCustomTextDialog.DialogOptions();
+            baseConf(activity, dopt);
+            dopt.isSearchEnabled = true;
+            dopt.titleText = R.string.file_encryption_password;
+            final boolean hasPassword = !TextUtils.isEmpty(as.getPassword());
+            dopt.messageText = hasPassword ? activity.getString(R.string.password_already_set_setting_a_new_password_will_overwrite) : "";
+            dopt.searchHintText = hasPassword ? R.string.hidden_password : R.string.empty_string;
+            dopt.callback = password -> {
+                if (!TextUtils.isEmpty(password)) {
+                    final String key = activity.getString(R.string.pref_key__default_encryption_password);
+                    new PasswordStore(activity).storeKey(password, key, PasswordStore.SecurityMode.NONE);
```

Comment:
Wouldn't it be besser to have an AppSettings.storePassword?

Reply:
I agree. Would you be able to push these two changes to this PR?

Reply:
@harshad1 I can't make changes to this PR, but I create a PR for your repo, so be so kindly an merge it into your branch.

Reply:
@opensource21 Thanks. Merged